src/main/java/seedu/duke/data/TimeTableManager.java

```diff
@@ -154,6 +164,37 @@ public static void deleteLesson(DayOfWeek dayOfWeek, int lessonIndex) throws Ind
         return lessonManager.getDayLessonList(dayOfWeek);
     }
 
+    /**
+     * Checks if the given lesson overlaps with any lesson already in timetable with the given repeat.
+     *
+     * @param lesson
+     *  The specified lesson
+     * @return
+     *  The lesson in timetable that the given lesson overlaps with, or null if no overlap
+     */
+    public static Lesson doesOverlap(Lesson lesson, int repeat) {
+        LessonManager lessonManager;
+        for (int i = getCurrWeekNum(); i < timetable.getSemEndWeekNum(); i++) {
+            if (repeat == 2 && !isEvenWeek(i)) {         // Handle even repeat
+                continue;
+            } else if (repeat == 3 && !isOddWeek(i)) {   // Handle odd repeat
+                continue;
+            }
+
+            lessonManager = timetable.getLessonManagerOfWeek(i);
+            for (Lesson eachLesson : lessonManager.getDayLessonList(lesson.getDay())) {
+                if (eachLesson.checkOverlap(lesson)) {
+                    return eachLesson;
+                }
+            }
+
+            if (repeat == 0) {  // Handle no repeat
+                return null;
+            }
```

Comment:
Maybe move this section to line 182 with the other if statements

Reply:
@Aseanseen Can't because it needs to check whether the current week (i.e. the only week it will be added to) has any overlaps.